boot/bootutil/src/bootutil_misc.c

```diff
@@ -580,6 +580,62 @@ boot_write_enc_key(const struct flash_area *fap, uint8_t slot,
 }
 #endif
 
+#ifdef MCUBOOT_XIP
+/**
+ * Extract the version number from the image header.
+ *
+ * @param hdr  Pointer to the image header structure.
+ *
+ * @note       This function must be ported if version number format has
+ *             changed in the image header.
+ *
+ * @return     Version number converted to uint64_t.
+ */
+uint64_t
+boot_get_version_number(struct image_header *hdr)
+{
+    uint64_t version = 0;
+
+    version |= (uint64_t)hdr->ih_ver.iv_major << (IMAGE_VER_MINOR_LENGTH
+                                                + IMAGE_VER_REVISION_LENGTH
+                                                + IMAGE_VER_BUILD_NUM_LENGTH);
+    version |= (uint64_t)hdr->ih_ver.iv_minor << (IMAGE_VER_REVISION_LENGTH
+                                                + IMAGE_VER_BUILD_NUM_LENGTH);
+    version |= (uint64_t)hdr->ih_ver.iv_revision << IMAGE_VER_BUILD_NUM_LENGTH;
+    version |= hdr->ih_ver.iv_build_num;
+    return version;
+}
+
+/**
+ * Comparator function for `qsort` to compare version numbers.
+ *
+ * @param ver1  Pointer to an array element which holds the 1. version number
+ * @param ver2  Pointer to an array element which holds the 2. version number
+ *
+ * @note        This function must be ported if version number format has
+ *              changed in the image header.
+ *
+ * @return      if version1 >  version2  -1
```

Comment:
This ordering seems a bit backward. Other functions that do comparisons (such as memcmp) will return -1 if the first value is less than the second.

Reply:
I realize this is to make the qsort below work, but it should at least be clearly stated that the comparison is reversed.

Reply:
this function has been removed (along with qsort)